collect_app/src/androidTest/java/org/odk/collect/android/support/pages/AsyncPage.kt

```diff
@@ -0,0 +1,11 @@
+package org.odk.collect.android.support.pages
+
+import org.odk.collect.testshared.WaitFor
+
+class AsyncPage<T : Page<T>>(private val destination: T) : Page<T>() {
+    override fun assertOnPage(): T {
```

Comment:
Wouldn't it be better (easier/safer) to add `waitFor` to each `assertOnPage` so that we don't need this wrapper?

Reply:
Good point! I did consider just adding `waitFor` to `Page` so that it runs on every `assertOnPage`, but that's a much more widespread change, obviously. I'll give it a try and see if it causes any problems.

Reply:
I think it'll take a bunch of busy work or rethinking to make every page assertion use `waitFor`. I suggest we go with this for the moment and then file an issue for making the `waitFor` the default.